task19/js/news.js

```diff
@@ -0,0 +1,157 @@
+    const menuListElement = createElementWithClassName("li", "menu-list");
+    menuListElement.dataset.index = i;
+    menuListElement.textContent = newsData[i].category;
+    newsData[i].display && menuListElement.classList.add("active");
+    tabMenuFragment.appendChild(menuListElement);
+  }
+  tab.appendChild(tabMenuFragment);
+};
+
+const renderTabContainer = (newsData) => {
+  const tabContentsFragment = document.createDocumentFragment();
+  for (let i = 0; i < newsData.length; i++) {
+    const newsWrapperElement = createElementWithClassName("div", "news-wrapper");
+    const articleTitlelistElement = createElementWithClassName(
+      "ul",
+      "article-title-list"
+    );
+    const newsImgElement = createElementWithClassName("img", "news-img");
+    newsData[i].display && newsWrapperElement.classList.add("show");
+    newsImgElement.src = newsData[i].img;
+
+    tabContentsFragment
+      .appendChild(newsWrapperElement)
+      .appendChild(articleTitlelistElement)
+      .appendChild(createArticleItem(newsData[i]));
+
+    tabContentsFragment.appendChild(newsWrapperElement).appendChild(newsImgElement);
+  }
+  const tabContainerElement = createElementWithClassName("div", "tab-container");
+  tabWrapper.appendChild(tabContainerElement).appendChild(tabContentsFragment);
+};
+
+const createArticleItem = ({ article }) => {
+  const articleTitleFragment = document.createDocumentFragment();
+  for (let i = 0; i < article.length; i++) {
+    const articleTitleItemElement = createElementWithClassName(
+      "li",
+      "article-title-item"
+    );
+    const articleWrapperElement = createElementWithClassName(
+      "article",
+      "article-wrapper"
+    );
+    const articleLinkElement = createElementWithClassName("a", "article-link");
+    const articleTitleElement = createElementWithClassName("h1", "article-title");
+    articleTitleElement.textContent = article[i].title;
+    articleLinkElement.href = "#";
+
+    articleTitleFragment
+      .appendChild(articleTitleItemElement)
+      .appendChild(articleWrapperElement)
+      .appendChild(articleLinkElement)
+      .appendChild(articleTitleElement);
+
+    renderNewArrivals(article[i], articleLinkElement);
+    renderCommentCount(article[i], articleLinkElement);
+  }
+  return articleTitleFragment;
+};
+
+const renderNewArrivals = (article, parent) => {
+  const diffDays = getDiffDays(article.date);
+  const newArrivalsLimiter = 3;
+  if (diffDays <= newArrivalsLimiter) {
+    parent.appendChild(createNewArrivals());
+  }
+};
+
+const renderCommentCount = (article, parent) => {
+  const commentLength = article.comment.length;
+  if (commentLength > 0) {
+    parent.appendChild(createCommentCount(commentLength));
+  }
+};
+
+const getDiffDays = (articlePostedDate) => {
+  const postedDate = format(new Date(articlePostedDate), "yyyy, MM, dd");
+  const today = format(new Date(), "yyyy, MM, dd");
+  const elapsed = differenceInCalendarDays(new Date(today), new Date(postedDate));
```

Comment:
nit
```suggestion
  const diffDays = differenceInCalendarDays(new Date(today), new Date(postedDate));
```

Reply:
Thank you so much for always correcting my name!!! It is very informative.
[91b6fb3](https://github.com/seeeen044/JavaScript_handson/pull/20/commits/91b6fb37450f1acca6d7b58b73f02b0b037fbe0b)